File: src/com/OpenSerenity/core/Browser.java
package com.OpenSerenity.core;

import com.OpenSerenity.elements.BaseElement;
import com.OpenSerenity.functionalInterfaces.Func;

import java.util.Set;

public interface Browser {
    <TElement extends BaseElement> TElement findElement(Class<TElement> clz, String locator, Func<Browser> selectFrame) throws IllegalAccessException, InstantiationException;

    public void start();
    public void stop();
    public void open(String url);
    public void close();
    public void resizeTo(int width, int height);
    public void maximize();
    public void back();
    public void refresh();
    public void selectFrame(String locator);

    public void selectWindow(String handle);
    public void closeWindow(String handle);
    public String getCurrentWindowHandle();
    public Set<String> getAllWindowsHandles();

    public String getTitle();
    public String getLocation();
    public String getMessageBoxText();
    public void acceptMessageBox();
    public void dismissMessageBox();

    public void captureEntirePageScreenShot(String filename);
    public void createCookie(String name, String value);
    public String getCookie(String name);
    public void deleteCookie(String name, String path);
    public void acceptAnyAlert();
}

File: src/scala/com/OpenSerenity/core/Browser.scala
package com.OpenSerenity.core

import scala.Predef.String
import com.OpenSerenity.elements.BaseElement

abstract trait Browser {
  def findElement[TElement <: BaseElement[TElement]](locator: String, selectFrame: () => Browser)(implicit m: scala.reflect.Manifest[TElement]): TElement;
  def start
  def stop
  def open(url: String)
  def close
  def resizeTo(width: Int, height: Int)
  def maximize
  def back
  def refresh
  def selectFrame(locator: String)
  def selectWindow(handle: String)
  def closeWindow(handle: String)
  def getCurrentWindowHandle: String
  def getAllWindowsHandles: scala.collection.mutable.Set[String]
  def getTitle: String
  def getLocation: String
  def getMessageBoxText: String
  def acceptMessageBox
  def dismissMessageBox
  def captureEntirePageScreenShot(filename: String)
  def createCookie(name: String, value: String)
  def getCookie(name: String): String
  def deleteCookie(name: String, path: String)
  def acceptAnyAlert
  def getLocalStorage: LocalStorage
}
